lib/kramdown/converter/html.rb

```diff
@@ -148,11 +148,10 @@ def convert_header(el, indent)
           attr['id'] = generate_id(el.options[:raw_text])
         end
 
-        if @options[:auto_ids] && @options[:header_links]
+        if @options[:header_links] && attr['id'].length > 0
```

Comment:
`attr['id'].length` will error out if there is no 'id' set. So better use something like `attr['id'].to_s.length > 0`.